src/App.jsx

```diff
@@ -13,46 +15,122 @@ export const goodsFromServer = [
   'Jam',
   'Garlic',
 ];
+const preparedGoods
+  = [...goodsFromServer]
+    .map((good, index) => ({ good, key: index }));
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_BY_ALPHABET = 'alpha';
+const SORT_BY_LENGTH = 'length';
+const REVERSE = 'reverse';
+const RESET = 'reset';
+
+export const App = () => {
+  const [visibleGoods, setVisibleGoods] = useState(preparedGoods);
+  const [resetButtonHidden, setResetButtonHidden] = useState(false);
+  const [sortField, setSortField] = useState('');
+
+  function getSortBy(products, sortBy) {
+    if (sortBy === SORT_BY_ALPHABET) {
+      // eslint-disable-next-line no-console
+      console.log(products)
+
+      return () => {
+        setResetButtonHidden(true);
+        setSortField(SORT_BY_ALPHABET);
+        setVisibleGoods([...products]
+          .sort((good1, good2) => good1.good.localeCompare(good2.good)));
+      };
+    }
+
+    if (sortBy === SORT_BY_LENGTH) {
+      return () => {
+        setResetButtonHidden(true);
+        setSortField(SORT_BY_LENGTH);
+        setVisibleGoods([...products]
+          .sort((good1, good2) => good1.good.length - good2.good.length));
+      };
+    }
+
+    return () => {
+      setResetButtonHidden(true);
+      setSortField(REVERSE);
+      setVisibleGoods([...visibleGoods].reverse());
+    };
+  }
+
+  const resetGoods = () => {
+    setSortField(RESET);
+    setResetButtonHidden(false);
+    setVisibleGoods([...preparedGoods]);
+  };
 
```

Comment:
you are complicating the logic too much
try to do the same as in the video that was before this task